fontconfig-sys/Cargo.toml

```
@@ -27,6 +27,11 @@ name = "fontconfig_sys"
 
 [dependencies]
 const-cstr = "0.3"
+dlib = "0.5.0"
+lazy_static = "1.4.0"
+
+[features]
+dlopen = []
```

I think `lazy_static` should be an optional dependency enabled by the `dlopen` feature.

```suggestion
dlopen = ["lazy_static"]
```

---

I had it that way initially but unfortunately that doesn't work with build.rs enabling the `dlopen` feature from the environment variable:
```
fontconfig-rs on  dlopen [!?] via 🦀 v1.58.1 
❯ RUST_FONTCONFIG_DLOPEN=on cargo build
   Compiling pkg-config v0.3.24
   Compiling cfg-if v1.0.0
   Compiling const-cstr v0.3.0
   Compiling fontconfig v0.3.0 (/home/be/sw/fontconfig-rs/fontconfig)
   Compiling libloading v0.7.3
   Compiling dlib v0.5.0
   Compiling yeslogic-fontconfig-sys v3.0.0 (/home/be/sw/fontconfig-rs/fontconfig-sys)
error[E0432]: unresolved imports `fontconfig::LIB`, `fontconfig::LIB_RESULT`
  --> fontconfig-sys/src/lib.rs:16:22
   |
16 | pub use fontconfig::{LIB, LIB_RESULT};
   |                      ^^^  ^^^^^^^^^^ no `LIB_RESULT` in `fontconfig`
   |                      |
   |                      no `LIB` in `fontconfig`

error[E0433]: failed to resolve: use of undeclared crate or module `lazy_static`
  --> fontconfig-sys/src/fontconfig.rs:24:1
   |
24 | lazy_static::lazy_static! {
   | ^^^^^^^^^^^ use of undeclared crate or module `lazy_static`

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `yeslogic-fontconfig-sys` due to 2 previous errors
```
I added a comment to Cargo.toml to document this.

---

Ahh that's a shame. I found these references too:

- https://github.com/rust-lang/cargo/issues/5499
- https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-cfg
  > Note that this does not affect Cargo's dependency resolution. This cannot be used to enable an optional dependency, or enable other Cargo features.